用户关注功能后端更新

genius-child-alice opened the pull request:
本次 PR 为 Feed 模块新增了 4 个核心功能，查看关注列表、粉丝列表、关注统计以及获取关注对象的最新视频内容，本次功能改进了查询方式，不涉及数据库索引与字段的创建或修改

## ✨ 新增功能

### 1. 获取用户关注列表
- **路由**: `GET /feed/following/list`
- **功能**: 分页获取指定用户的关注列表
- **特性**:
  - 支持隐私权限控制（public/private/following）
  - 返回用户基本信息（uid, uuid, username, nickname, avatar）
  - 返回关注时间（followingCreateTime）
  - 支持分页查询（num, offset）
  - 返回关注总数（totalCount）

### 2. 获取用户粉丝列表
- **路由**: `GET /feed/follower/list`
- **功能**: 分页获取指定用户的粉丝列表
- **特性**:
  - 支持隐私权限控制（public/private/following）
  - 返回用户基本信息（uid, uuid, username, nickname, avatar）
  - 如果查看者已登录，会显示是否已关注该粉丝（isFollowing）
  - 支持分页查询（num, offset）
  - 返回粉丝总数（totalCount）

### 3. 获取关注统计
- **路由**: `GET /feed/stats`
- **功能**: 获取用户的关注数和粉丝数统计
- **特性**:
  - 返回关注数量（followingCount）
  - 返回粉丝数量（followerCount）
  - 无需认证即可访问（公开接口）

### 4. 获取关注对象的最新视频
- **路由**: `GET /feed/following/videos`
- **功能**: 获取当前用户关注的所有用户的最新视频
- **特性**:
  - 需要用户认证
  - 按上传时间降序排序（最新在前）
  - 支持分页查询（num, offset）
  - 返回视频详细信息（videoId, title, image, uploadDate, watchedCount, uploader, duration, description 等）
  - 返回视频总数（videosCount）

## 🔧 技术实现

### 数据库变更
**本次更新没有对数据库结构做任何修改**，所有功能都基于现有的数据库集合：
- `following` 集合 - 存储用户关注关系（FollowingSchema）
- `user-infos` 集合 - 存储用户基本信息（UserInfoSchema）
- `user-settings` 集合 - 存储用户隐私设置（UserSettingsSchema）
- `videos` 集合 - 存储视频数据（VideoSchema）

所有新增功能都是**只读查询**，不会修改数据库结构或新增字段。

### 隐私权限控制
通过 `checkPrivacyPermission` 函数，支持三种隐私级别：
- **public**: 所有人可见
- **private**: 仅自己可见
- **following**: 仅关注者可见

该函数通过查询 `user-settings` 集合中的 `userPrivaryVisibilitiesSetting` 字段来判断权限。

### 数据库查询优化（查询方式优化，非数据库结构变更）
本次更新采用了高效的查询方式，但**没有修改数据库结构或新增索引**：

1. **MongoDB Aggregation Pipeline**
   - 使用聚合管道进行复杂查询，减少多次数据库往返
   - 示例：关注列表查询使用 `$match` → `$sort` → `$skip` → `$limit` → `$lookup` → `$unwind` → `$project`

2. **`$lookup` 关联查询**
   - 在数据库层面关联 `user-infos` 集合，一次性获取用户信息
   - 避免在应用层进行多次查询

3. **`$count` 统计优化**
   - 使用聚合管道的 `$count` 阶段快速统计总数
   - 比 `countDocuments()` 更高效

4. **批量查询优化**
   - 粉丝列表中的 `isFollowing` 状态通过 `$in` 操作符批量查询
   - 一次查询获取所有关注关系，而不是循环查询

### 错误处理
- 完善的参数验证
- 详细的错误日志记录
- 友好的错误提示信息

## 📝 API 变更

### 新增路由

```typescript
// 获取关注列表
GET /feed/following/list?targetUid={uid}&num={num}&offset={offset}

// 获取粉丝列表
GET /feed/follower/list?targetUid={uid}&num={num}&offset={offset}

// 获取关注统计
GET /feed/stats?targetUid={uid}

// 获取关注对象最新视频
GET /feed/following/videos?num={num}&offset={offset}
```

### 请求参数

#### 关注/粉丝列表
- `targetUid` (number, 必填): 目标用户 UID
- `num` (number, 必填): 返回数量（1-200）
- `offset` (number, 必填): 偏移量（>= 0）

#### 关注统计
- `targetUid` (number, 必填): 目标用户 UID

#### 关注对象最新视频
- `num` (number, 必填): 返回数量（1-200）
- `offset` (number, 必填): 偏移量（>= 0）

### 响应格式

#### 关注列表响应
```typescript
{
  success: boolean;
  message?: string;
  totalCount?: number;
  result?: UserInfoForFollowList[];
}

// UserInfoForFollowList
{
  uid: number;
  uuid: string;
  username?: string;
  userNickname?: string;
  avatar?: string;
  followingCreateTime?: number;
}
```

#### 粉丝列表响应
```typescript
{
  success: boolean;
  message?: string;
  totalCount?: number;
  result?: UserInfoForFollowList[];
}

// UserInfoForFollowList (粉丝列表)
{
  uid: number;
  uuid: string;
  username?: string;
  userNickname?: string;
  avatar?: string;
  isFollowing?: boolean; // 查看者是否关注了该粉丝
}
```

#### 关注统计响应
```typescript
{
  success: boolean;
  message?: string;
  followingCount?: number;
  followerCount?: number;
}
```

#### 关注对象最新视频响应
```typescript
{
  success: boolean;
  message?: string;
  videosCount: number;
  videos: VideoInfoForFollowingList[];
}

// VideoInfoForFollowingList
{
  videoId: number;
  title: string;
  image?: string;
  uploadDate?: number;
  watchedCount?: number;
  uploader?: string;
  uploaderNickname?: string;
  uploaderId?: number;
  duration?: number;
  description?: string;
  isBlockedByOther?: boolean;
}
```

## 🔒 安全特性

1. **隐私权限控制**: 关注列表和粉丝列表支持隐私设置，根据用户设置决定是否可见
2. **用户认证**: 部分接口需要用户认证（通过 cookie 中的 uuid 和 token）
3. **参数验证**: 所有接口都有完善的参数验证，防止非法请求
4. **权限检查**: 使用 RBAC 系统进行权限验证

## 🧪 测试建议

- [ ] 测试关注列表的分页功能
- [ ] 测试粉丝列表的隐私权限控制
- [ ] 测试关注统计的准确性
- [ ] 测试关注对象最新视频的排序和分页
- [ ] 测试隐私设置的三种模式（public/private/following）
- [ ] 测试未登录用户访问公开接口
- [ ] 测试参数验证和错误处理

## 📚 相关文件

### 新增/修改的文件
- `src/controller/FeedController.ts` - 新增 4 个控制器函数
- `src/service/FeedService.ts` - 新增 4 个服务函数和隐私权限检查函数
- `src/controller/FeedControllerDto.ts` - 新增请求/响应 DTO 类型定义
- `src/route/router.ts` - 新增 4 个路由

## 🔗 相关 Issue

Closes #[issue-number]

## 📸 截图/示例

### 示例请求

```bash
# 获取用户关注列表
curl "https://api.example.com/feed/following/list?targetUid=100&num=20&offset=0"

# 获取用户粉丝列表
curl "https://api.example.com/feed/follower/list?targetUid=100&num=20&offset=0"

# 获取关注统计
curl "https://api.example.com/feed/stats?targetUid=100"

# 获取关注对象最新视频
curl -H "Cookie: uuid=xxx; token=xxx" "https://api.example.com/feed/following/videos?num=20&offset=0"
```

## ✅ 检查清单

- [x] 代码符合项目规范
- [x] 添加了必要的注释和文档
- [x] 实现了错误处理和日志记录
- [x] 支持隐私权限控制
- [x] 优化了数据库查询性能
- [x] 添加了参数验证
- [x] 更新了路由配置

genius-child-alice committed:
Submit feed related changes

genius-child-alice committed:
feat(动态): 添加获取关注对象最新视频功能

添加新的接口/feed/following/videos用于获取用户关注对象的最新视频列表
实现从数据库查询关注用户的视频并按时间排序返回
包含分页参数和视频信息完整字段

genius-child-alice committed:
fix(FeedController): 修复获取关注视频时的参数验证逻辑

将RBAC权限验证改为基础参数验证，确保请求包含必需的uuid和token参数

cfdxkk commented:
<img width="1051" height="149" alt="image" src="https://github.com/user-attachments/assets/a140f92a-305c-4a83-979f-9725c2c653b6" />

应该是 `admin` 角色

cfdxkk commented:
期望可以仅通过 RBAC 对每个接口的访问控制达成审核权限控制，不再需要 `checkUserHasXXXXXXRole` 这样的 '静态' 校验函数。

还没有仔细阅读代码，仅阅读了说明文档，如果确实需要这个校验函数，可以讨论。

genius-child-alice commented:
> 期望可以仅通过 RBAC 对每个接口的访问控制达成审核权限控制，不再需要 `checkUserHasXXXXXXRole` 这样的 '静态' 校验函数。
> 
> 还没有仔细阅读代码，仅阅读了说明文档，如果确实需要这个校验函数，可以讨论。

不好意思，我pr文档貌似写错了，这个是关注者方面的功能，审核系统交到alice分支里面了@cfdxkk

genius-child-alice committed:
feat: 添加日志服务并替换多处console输出为logging

refactor(i18n): 标准化客户端语言标识符处理逻辑

feat(DbClusterPoolTypes): 扩展UpdateType支持$inc操作符

feat(locales): 更新多语言邮件模板并添加新验证码类型

fix: 修复多处错误处理和日志记录

feat: 新增DateTool工具函数isToday

feat: 新增LoggingSchema用于日志存储

refactor: 优化错误处理和信息输出

docs: 更新UserControllerDto类型定义

style: 清理代码并统一格式

chore: 更新依赖和配置

YasaitoriOvO commented on src/route/router.ts:
```diff
@@ -746,6 +718,22 @@ router.get('/feed/getFeedContent', getFeedContentController) // 获取动态分
+router.get('/feed/following/videos', getFollowingVideosController) // 获取我关注对象的最新视频
+// https://localhost:9999/feed/following/videos?num=20&offset=0
+// cookie: uuid, token
+
```
```Query``` 也要写一下

同时注释中的 请求参数 要与实际相同

YasaitoriOvO commented on src/controller/FeedController.ts:
```diff
@@ -278,3 +278,131 @@ export const getFeedContentController = async (ctx: koaCtx, next: koaNext) => {
+
+	ctx.body = await getFollowStatsService(getFollowStatsRequest, uuid, token)
+	await next()
+}
+
+/**
+ * 获取我关注对象的最新视频
+ * @param ctx context
+ * @param next context
+ * @return 获取我关注对象的最新视频的请求响应
+ */
+export const getFollowingVideosController = async (ctx: koaCtx, next: koaNext) => {
+	const uuid = ctx.cookies.get('uuid')
+	const token = ctx.cookies.get('token')
+	
+	const numStr = ctx.query.num as string
+	const offsetStr = ctx.query.offset as string
+
+	const num = parseInteger(numStr)
+	const offset = parseInteger(offsetStr)
+
+	if (!num || offset === undefined) {
+		ctx.body = { success: false, message: '参数不合法', videosCount: 0, videos: [] }
+		return
+	}
+
+	// 验证用户身份（uuid 和 token 在 service 层验证）
+	if (!uuid || !token) {
+		ctx.body = { success: false, message: '参数不合法：缺少 uuid 或 token', videosCount: 0, videos: [] }
+		return
+	}
+
+	const getFollowingVideosRequest: GetFollowingVideosRequestDto = {
+		num,
+		offset,
+	}
+
+	ctx.body = await getFollowingVideosService(getFollowingVideosRequest, uuid, token)
+	await next()
+}
```
变量只需要一个 然后可以根据别的Controller来进行修改

YasaitoriOvO commented on src/service/FeedService.ts:
```diff
@@ -996,3 +998,619 @@ const checkGetFeedContentRequest = (getFeedContentRequest: GetFeedContentRequest
+		getFollowStatsRequest.targetUid !== null &&
+		getFollowStatsRequest.targetUid > 0
+	)
+}
```
验证的建议都放在末尾
return 在较短的情况下可以不用换行

genius-child-alice committed:
🎨改进代码的结构和格式

cfdxkk commented on src/service/FeedService.ts:
```diff
@@ -893,11 +895,482 @@ export const getFeedContentService = async (getFeedContentRequest: GetFeedConten
+
+		const result: UserInfoForFollowList[] = (listResult.result || []).map((item: any) => ({
+			uid: item.uid || 0,
+			uuid: item.uuid || '',
```
UUID 只用作本人登录验证和内部关联数据。

genius-child-alice committed:
Merge remote-tracking branch 'origin/develop' into feed

cfdxkk commented on .github/workflows/build-docker.yml:
怎么给这个文件删掉了，你再同步一下最新的 develop 试试？

下面还有好几个类似的文件也有问题。

cfdxkk commented on .vscode/tasks.json:
```diff
@@ -21,7 +21,7 @@
 			"problemMatcher": [],
 			"label": "npm: dev-hot",
-			"detail": "npm dev-hot - 启动本地后端开发服务器，并启用自动重载"
+			"detail": "npm dev-hot - 启动本地后端开发热重载服务器"
```
这个地方也是给 develop 的覆盖了

genius-child-alice replied:
已修复

genius-child-alice committed:
🐛 移除 UserInfoForFollowList 中的 uuid 字段

重构关注列表相关服务，不再返回 uuid 字段。内部逻辑仍使用 uuid 进行关注关系检查，但最终返回数据中移除此字段以简化接口。

cfdxkk commented on src/controller/FeedController.ts:
```diff
@@ -278,3 +278,131 @@ export const getFeedContentController = async (ctx: koaCtx, next: koaNext) => {
+	if (!targetUid || !num || offset === undefined) {
+		ctx.body = { success: false, message: '参数不合法' }
+		return
+	}
```
参数验证习惯放在 service 层并单独放到一个函数里

genius-child-alice replied:
已修改

cfdxkk replied:
还没改完。
<img width="2732" height="2048" alt="IMG_0332" src="https://github.com/user-attachments/assets/0d840808-b1be-4dc0-8ce9-a6827be79c84" />
<img width="2732" height="2048" alt="IMG_0331" src="https://github.com/user-attachments/assets/cd84fbd0-3f8a-4264-85b6-a82af98aca20" />

cfdxkk commented on src/controller/FeedController.ts:
建议修改分页相关变量的命名与组织方式以符合以往的风格

genius-child-alice replied:
已修改

genius-child-alice committed:
鎭㈠ .github/workflows/build-docker.yml 骞跺悓姝?.vscode/tasks.json 鍒?develop 鐗堟湰

cfdxkk commented on src/service/FeedService.ts:
```diff
@@ -899,6 +900,477 @@ export const getFeedContentService = async (getFeedContentRequest: GetFeedConten
+					uploaderId: 1,
+					duration: 1,
+					description: 1,
+					uploader: '$uploader_info.username',
+					uploaderNickname: '$uploader_info.userNickname',
+				},
+			},
+		]
+		const listResult = await selectDataByAggregateFromMongoDB(videoSchemaInstance, videoCollectionName, listPipeline)
+
+		if (!listResult.success) {
+			logging('ERROR', '获取关注对象最新视频失败：查询失败')
+			return { success: false, message: '获取关注对象最新视频失败：查询失败', videosCount: 0, videos: [] }
+		}
+
+		const videos: GetFollowingVideosResponseDto['videos'] = (listResult.result || []).map((item: any) => ({
+			videoId: item.videoId || 0,
+			title: item.title || '',
+			image: item.image,
+			uploadDate: item.uploadDate,
+			watchedCount: item.watchedCount,
+			uploader: item.uploader,
+			uploaderNickname: item.uploaderNickname,
+			uploaderId: item.uploaderId,
+			duration: item.duration,
+			description: item.description,
+			isBlockedByOther: false, // 这里可以根据需要添加屏蔽检查
+		}))
+
+		return {
+			success: true,
+			message: '获取关注对象最新视频成功',
+			videosCount: totalCount,
+			videos,
+		}
+	} catch (error) {
+		logging('ERROR', '获取关注对象最新视频失败：未知错误', error)
+		return { success: false, message: '获取关注对象最新视频失败：未知错误', videosCount: 0, videos: [] }
+	}
+}
```
```suggestion
```

“动态”功能由动态分组实现

genius-child-alice committed:
🎨 添加验证获取用户关注列表参数的函数

将参数验证逻辑提取到单独的函数中，提高代码复用性和可维护性

genius-child-alice committed:
🎨 将offset/num分页方式改为page/pageSize方式

重构分页参数结构，将原有的offset/num分页方式改为更符合业务逻辑的page/pageSize方式
添加分页参数验证方法，统一校验page和pageSize的合法性
更新相关DTO定义和控制器逻辑，保持接口行为不变

genius-child-alice committed:
✨ 移除获取关注对象最新视频功能

删除不再使用的获取关注对象最新视频相关代码，包括控制器、服务、路由和DTO定义

cfdxkk commented:
你去这个 pr 的 changed files 里看看，把那些覆盖了 develop 分支修改的都解决掉。
https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Rosales/pull/88/changes

cfdxkk commented on src/service/FeedService.ts:
```diff
@@ -899,6 +900,349 @@ export const getFeedContentService = async (getFeedContentRequest: GetFeedConten
+			{
+				$project: {
+					uid: '$userInfo.uid',
+					uuid: '$followingUuid',
```
不能暴露 UUID，所有地方都需要注意。
虽然下面有遍历处理返回值，但这里也不需要返回 UUID

cfdxkk commented on src/service/FeedService.ts:
```diff
@@ -899,6 +900,349 @@ export const getFeedContentService = async (getFeedContentRequest: GetFeedConten
+				if (followingResult.success && followingResult.result) {
+					const followingUuidSet = new Set(followingResult.result.map(f => f.followingUuid))
+					result = tempResult.map(r => {
+						const { uuid, ...rest } = r
```
不要返回 UUID，这个地方逻辑有点复杂，仔细review一下

genius-child-alice committed:
🎨 添加验证获取关注统计参数的功能

添加 validateGetFollowStatsParams 函数用于验证获取用户关注数和粉丝数的参数，并在控制器中使用该验证逻辑

genius-child-alice committed:
refactor(FeedService): 移除前端不需要的uuid字段并重命名内部使用字段

将follower列表中的uuid字段重命名为_followerUuid以明确其内部使用属性，并移除following列表中的uuid字段，因为这些字段不需要返回给前端。同时更新相关内部逻辑以使用新命名字段。

YasaitoriOvO commented on src/controller/FeedController.ts:
```diff
@@ -278,3 +278,103 @@ export const getFeedContentController = async (ctx: koaCtx, next: koaNext) => {
+
+	const targetUid = parseInteger(targetUidStr)
+	const page = parseInteger(pageStr)
+	const pageSize = parseInteger(pageSizeStr)
```
同 ```im-system```

YasaitoriOvO commented on src/controller/FeedController.ts:
```diff
@@ -278,3 +278,103 @@ export const getFeedContentController = async (ctx: koaCtx, next: koaNext) => {
+	const targetUid = parseInteger(targetUidStr)
+	const page = parseInteger(pageStr)
+	const pageSize = parseInteger(pageSizeStr)
+
```
同 ```im-system```

YasaitoriOvO commented on src/controller/FeedController.ts:
```diff
@@ -278,3 +278,103 @@ export const getFeedContentController = async (ctx: koaCtx, next: koaNext) => {
+	const getFollowStatsRequest: GetFollowStatsRequestDto = {
+		targetUid: targetUid!,
+	}
+
```
可以试着学学其他文件的风格

YasaitoriOvO commented on src/controller/FeedController.ts:
```diff
@@ -278,3 +278,103 @@ export const getFeedContentController = async (ctx: koaCtx, next: koaNext) => {
+	const page = parseInteger(pageStr)
+	const pageSize = parseInteger(pageSizeStr)
+
+	const validationResult = validateGetFollowingListParams(targetUid, page, pageSize)
```
不用在这里进行校验

YasaitoriOvO commented on src/controller/FeedController.ts:
```diff
@@ -278,3 +278,103 @@ export const getFeedContentController = async (ctx: koaCtx, next: koaNext) => {
+	const page = parseInteger(pageStr)
+	const pageSize = parseInteger(pageSizeStr)
+
+	const validationResult = validateGetFollowerListParams(targetUid, page, pageSize)
```
同上

YasaitoriOvO commented on src/controller/FeedController.ts:
```diff
@@ -278,3 +278,103 @@ export const getFeedContentController = async (ctx: koaCtx, next: koaNext) => {
+	const targetUidStr = ctx.query.targetUid as string
+	const targetUid = parseInteger(targetUidStr)
+
+	const validationResult = validateGetFollowStatsParams(targetUid)
```
同上

YasaitoriOvO commented on src/service/FeedService.ts:
```diff
@@ -899,6 +900,348 @@ export const getFeedContentService = async (getFeedContentRequest: GetFeedConten
+			message: '获取关注列表成功',
+			totalCount,
+			result,
+		}
```
建议是改为单行

genius-child-alice committed:
refactor(FeedController): 重构分页参数为对象形式

将分页参数 page 和 pageSize 重构为 pagination 对象形式，提高代码可读性和维护性

YasaitoriOvO commented on src/service/FeedService.ts:
```diff
@@ -997,3 +1340,177 @@ const checkGetFeedContentRequest = (getFeedContentRequest: GetFeedContentRequest
+			return true
+		}
+
+		const visibilityType = privacySetting.visibilitiesType
+
+		// public: 所有人能看
+		if (visibilityType === 'public') {
+			return true
+		}
+
+		// private: 隐藏，只有自己能看
+		if (visibilityType === 'private') {
+			return false
+		}
+
+		// following: 仅关注能看，需要检查查看者是否关注了目标用户
+		if (visibilityType === 'following') {
+			if (!viewerUuid) {
+				return false
+			}
+			const { collectionName: followingCollectionName, schemaInstance: followingSchemaInstance } = FollowingSchema
+			type Following = InferSchemaType<typeof followingSchemaInstance>
+			const followingWhere: QueryType<Following> = {
+				followerUuid: viewerUuid,
+				followingUuid: targetUuid,
+			}
+			const followingSelect: any = {
+				_id: 1,
+			}
+			const followingResult = await selectDataFromMongoDB<Following>(followingWhere, followingSelect, followingSchemaInstance, followingCollectionName)
+			return followingResult.success && followingResult.result && followingResult.result.length > 0
+		}
+
+		// 默认不允许
+		return false
+	} catch (error) {
+		logging('ERROR', '检查隐私权限失败：', error)
+		return false
+	}
+}
```
这里不用返回一个 ```json``` 之类的
而且不需要导出 仅在 ```service层``` 内调用
直接返回一个布尔值就好了

cfdxkk replied:
注意原来的 checkXxxxx 系列函数中的 return 部分不必（且反对）按以前的风格写，以前的风格很不好。
```
// Not to be
return (
	!!removeUidFromFeedGroupRequest.feedGroupUuid
	&& !!removeUidFromFeedGroupRequest.uidList && removeUidFromFeedGroupRequest.uidList.every(uid => uid !== undefined && uid !== null && uid > 0)
)
```

`&&` 放到后面是对的，我选择放在前面是因为 `!!` 在行首会被识别为警告，导致一整行被高亮。
实际上应该修改一下 VSCode 插件配置，但我想改时已经来不及了。

genius-child-alice committed:
🎨 简化返回对象的格式

移除返回对象中的多余换行和缩进，保持代码风格一致

YasaitoriOvO committed:
🎨 QoL: feed相关业务

YasaitoriOvO committed:
Merge branch 'feed' of https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Rosales into feed

YasaitoriOvO committed:
🎨 QoL: Feed

YasaitoriOvO reviewed (changes requested):
希望可以重构一下整个 ```Service层``` 可以尽量按照以前的风格重构

cfdxkk commented:
我觉得现在这个还行，把验证搬到 service， 再把那几个验证函数的输入和返回类型改一改，就没什么异味了。

genius-child-alice committed:
🎨 优化关注和粉丝相关功能，重构相关请求和响应格式，增加关注数和粉丝数统计接口

cfdxkk commented on src/controller/FeedController.ts:
```diff
@@ -278,3 +278,77 @@ export const getFeedContentController = async (ctx: koaCtx, next: koaNext) => {
+	const targetUid = parseInteger(ctx.query.targetUid)
+	const page = ctx.query.page as string
+	const pageSize = ctx.query.pageSize as string
+	const finalPageSize =limitPageSize(pageSize)
```
```suggestion
	const finalPageSize = limitPageSize(pageSize)
```

YasaitoriOvO commented:
> 我觉得现在这个还行，把验证搬到 service， 再把那几个验证函数的输入和返回类型改一改，就没什么异味了。

我主要是觉得pipeline那一块和以前的风格差距好像有点大

cfdxkk reviewed (dismissed):
当前的 PR 保留了最基础的功能，去掉了查看粉丝／关注列表中的用户是否已被自己关注的功能。

cfdxkk commented:
> > 我觉得现在这个还行，把验证搬到 service， 再把那几个验证函数的输入和返回类型改一改，就没什么异味了。
> 
> 我主要是觉得pipeline那一块和以前的风格差距好像有点大

Pipeline 本身是自由的，但使用 pipeline 前提是足够理解业务。

YasaitoriOvO committed:
Update src/controller/FeedController.ts

Co-authored-by: cfdxkk <b2567240058@gmail.com>
